src/consumption.cpp

```diff
@@ -1858,17 +1855,14 @@ bool player::consume( item &target, bool force, item_pocket *const parent_pocket
     if( is_player() && !query_consume_ownership( target, *this ) ) {
         return false;
     }
-    if( consume_med( comest, *this, parent_pocket ) ||
-        eat( comest, *this, force, parent_pocket ) ||
-        feed_reactor_with( comest, parent_pocket ) ||
-        feed_furnace_with( comest, parent_pocket ) ||
-        fuel_bionic_with( comest, parent_pocket ) ) {
-
-        if( &target != &comest ) {
-            target.on_contents_changed();
-        }
+    if( consume_med( target, *this, parent_pocket ) ||
+        eat( target, *this, force, parent_pocket ) ||
+        feed_reactor_with( target, parent_pocket ) ||
+        feed_furnace_with( target, parent_pocket ) ||
+        fuel_bionic_with( target, parent_pocket ) ) {
 
-        return comest.charges <= 0;
+        target.on_contents_changed();
```

Comment:
Some of the calls in the if condition only returns true if the item is completely consumed (`consume_med` for example).

Reply:
Will look at that. Thanks for the feedback.